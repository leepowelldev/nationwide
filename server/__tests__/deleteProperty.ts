import { gql } from 'apollo-server';
import { mockCollection } from './__utils__/mockCollections.js';
import setup from './__utils__/setup.js';

const query = gql`
  mutation ($id: ID!) {
    deleteProperty(id: $id)
  }
`;

describe('Mutation: deleteProperty', () => {
  test('should delete property', async () => {
    const { server, collection } = setup(mockCollection);
    const id = '1';

    // Check collection has default size
    expect(collection.size).toStrictEqual(mockCollection.size);

    const res = await server.executeOperation({
      query,
      variables: {
        id,
      },
    });

    // Check collection has changed size
    expect(collection.size).toStrictEqual(0);
    expect(res).toMatchSnapshot();
  });

  test('should error if property does not exist', async () => {
    const { server, collection } = setup(mockCollection);
    const id = '2';

    // Check collection has default size
    expect(collection.size).toStrictEqual(1);

    const res = await server.executeOperation({
      query,
      variables: {
        id,
      },
    });

    // Check collection has not changed size
    expect(collection.size).toStrictEqual(1);
    expect(JSON.stringify(res.errors?.[0])).toMatchSnapshot();
    expect(res).toMatchSnapshot();
  });
});
