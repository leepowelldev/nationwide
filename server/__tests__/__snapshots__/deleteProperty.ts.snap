// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mutation: deleteProperty should delete property 1`] = `
Object {
  "data": Object {
    "deleteProperty": "1",
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Mutation: deleteProperty should error if property does not exist 1`] = `"{\\"message\\":\\"property with id 2 does not exist\\",\\"locations\\":[{\\"line\\":2,\\"column\\":3}],\\"path\\":[\\"deleteProperty\\"],\\"extensions\\":{\\"code\\":\\"NOT_FOUND\\"}}"`;

exports[`Mutation: deleteProperty should error if property does not exist 2`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: property with id 2 does not exist],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
